prophy/generators.py

```diff
@@ -29,6 +29,69 @@ def _build_up_implementation(cls):
         """
             Implementation of type creation. To be overridden in derived metaclasses.
         """
+    def __eq__(cls, other):
+        if not isinstance(other, cls.__class__):
+            return NotImplemented
+
+        if cls.__bases__ != other.__bases__:
+            return False
+
+        for cls_d, other_d in zip(cls._descriptor, other._descriptor):
```

Comment:
You need to also check both `_descriptor`'s lenghts first. Result of zip has length of the shortest iterable passed.